src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -48,11 +45,21 @@
       + "watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|"
       + "embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
 
+  @VisibleForTesting static final String PARAM_ID = "id";
+
   /* Name of input field used for lecture name in lecture selection page. */
-  private static final String NAME_INPUT = "name-input";
+  @VisibleForTesting static final String PARAM_NAME = "name-input";
 
   /* Name of input field used for lecture video link in lecture selection page. */
-  private static final String LINK_INPUT = "link-input";
+  @VisibleForTesting static final String PARAM_LINK = "link-input";
+
+
+  @VisibleForTesting static final String ERROR_MISSING_NAME = "Missing name parameter.";
+  @VisibleForTesting static final String ERROR_MISSING_LINK = "Missing link parameter.";
+  @VisibleForTesting static final String ERROR_MISSING_ID = "Missing id parameter.";
+  @VisibleForTesting static final String ERROR_INVALID_LINK = "Invalid video link.";
+  @VisibleForTesting static final String ERROR_FAILED_REDIRECT = "Failed to create redirect URL.";
+  @VisibleForTesting static final String ERROR_LECTURE_NOT_FOUND = "Lecture not found in database.";
```

Comment:
Ditto

Reply:
"If you expose constants in Java, please mark the modifications with @VisibleForTesting"
This is from the doc: https://docs.google.com/document/d/18T9ef-DRqTncVD377GoIQmOGI84cNKkZ7Nn07YURFTw/edit